### DISCUSSION TITLE
A better way to display the note-map?

### DISCUSSION THREAD

**Discussion by soulsands** [TriliumNext/Trilium#3881] [General]
Hello.
I have been using Trilium for several months and really enjoy it.
The note-map of the tree view is like an expanded mind map, which can provide some intuitive information. However, the note-map of the link view can provide very limited information. I think the parent-child relationship of notes is also an important link that should be included and displayed together with the links.
Back to our usage scenario, there should be two types of scenarios that can help us.1. View all notes related to the current note, which can help us discover as many associations as possible. 2. Find out which group of related notes exist among many notes, thereby discovering unexpected connections.
![image](https://user-images.githubusercontent.com/32272399/235076331-3eb2307e-88f3-4135-bb29-2eef9e8b6dee.png)
In the example above, I also included the parent-child relationship of the notes, displayed along with the links. The number of strength for the parent-child relationship can be set, in this case it is 2 layers, showing the current note's parent, grandparent, children, grandchildren, and siblings, personally thinking this kind of presentation is more valuable.

I have seen similar discussions, https://github.com/zadam/trilium/issues/3840. 
What do you think?

**Comment by zadam** [comment]
Hi, my position is basically described in #3840 - IIRC I tried including both, but on real world data it looked bad.

But maybe I was wrong, overlooked something or there is some solution, perhaps playing with relative strengths.

**Reply by soulsands** [reply]
@zadam  Thanks for replying. "Relative strength" is the concept in my mind. I think relatives should be considered as links and they can work together with former links. Relatives can be differentiated by using a different color in the frontend display. In the above example, I used a lighter color. Alternatively, different colors could be used based on the strength of the relationship.
I have read the book [How to Take Smart Notes](https://www.amazon.com/How-Take-Smart-Notes-Nonfiction/dp/1542866502),from which I got some inspirations. You may be willing to take a look at it, if you haven't read yet.

Back to note-map, do you agree with this approach? if so, we can make more discussions on it.

**Reply by zadam** [reply]
> Back to note-map, do you agree with this approach? 

Maybe? IMHO it's a matter of experimentation. If it improves the usefulness of the note map I'm happy to accept, but I can't say that before I can test it on some larger data sets.

**Reply by soulsands** [reply]
Okay, I understand. My implementation is simple: traverse the parents and children of a note, save the result, and transform them into links with a relative strength property. Original links and relatives can be distinguished by colors on the frontend. However, I am unsure whether the relative strength flag should be configurable since it could cause severe performance problems. Besides, it may has some conflicts with on going next60, so I think it might be better to work on this after 60 released.
